### ISSUE TITLE
Top JavaScript errors reported by Track.js

### ISSUE THREAD

**Issue by ghost** [freeciv/freeciv-web#119]
These are the top JavaScript errors that are reported by Track.js:
![bilde](https://user-images.githubusercontent.com/210993/29489903-f96ae634-852b-11e7-992d-1f89c9a508e8.png)

I can provide more information about each of these, and also give login-information to Track.js for Freeciv-web contributors.

**Comment by lonemadmax** [comment]
The second one looks like a syntax problem perfect for someone new to the project.

**Comment by andreasrosdal** [comment]
Yes, unfortunately there is very little information in Track.js about where that particular bug occurs.
Send me an e-mail to andrearo@pvv.ntnu.no and I can give you access to Track.js.

**Comment by cpetry** [comment]
Could you create an issue for each of these?
Perhaps with some more information about each bug.

**Comment by andreasrosdal** [comment]
@cpetry Please send me your e-mail address to andrearo@pvv.ntnu.no and I will give you access to Track.js.

**Comment by cpetry** [comment]
I never worked with trace.js before. It seems I would need a sourcemap to be able to pinpoint some errors.
Where/how do I get/create this for the js code?

**Comment by lonemadmax** [comment]
IIUC, the sourcemap is a mapping from tokens in the minified code to the ones in the original source.  When building freeciv-web it is generated in freeciv-web/target (I think, I'm not at home now and I can't check). It's also in https://play.freeciv.org/javascript/webclient.min.js.map. I guess either track.js detects it automatically or Andreas uploads it, as I can see it applied and I have done nothing.  Still, I prefer to set it to off and "See Raw Source".  The minification is only local and usually easy to recognize, and applying the map just gives you the name of the token. I'm also new to track.js, so maybe there's somewhere I haven't clicked yet to get more info.

A different thing is code in libraries that are used already minified, those won't appear in that map. Maybe their projects provide the corresponding maps, I haven't checked.

For example, in "Unable to load tileset spec. Run Freeciv-img-extract." I can see the map applied to webclient.js, but not to jquery.

And then, there are some errors without stack info, like the "Unexpected token var" one.

Also, I've seen errors from translate.googleusercontent.com, probably someone trying to open freeciv-web from inside google translator. So not all the errors you see come directly because of a coding error in frreciv-web.

**Comment by cpetry** [comment]
Thanks for the detailed help!
I just wondered that I got no information on the "Unexpected token var" error.
I'll try to fix some errors that have hints about where to locate them.

**Comment by lonemadmax** [comment]
Re the Unexpected token var errors, I've noticed now (I don't know if it is something new or I was blind before) that they say they belong to a "file" included as a running base64 insertion. It seems to belong to the urchin tracker.

**Comment by mchenryc** [comment]
> and also...  login-information to Track.js for Freeciv-web contributors.

@lonemadmax I'm interested in read access, if the offer above is still open.

**Comment by lonemadmax** [comment]
Pinging @andreasrosdal

**Comment by andreasrosdal** [comment]
what is your email?

**Comment by mchenryc** [comment]
mchenryc@gmail.com

**Comment by andreasrosdal** [comment]
invitation sent

**Comment by andreasrosdal** [comment]
delete your email comment to prevent spam
